### PR TITLE
nitweb: use doc-commands

### PR DESCRIPTION
**Pull request by Morriar**
Migrate nitweb to the doc-commands system.

Changes:
* moved the group to `doc::api`
* use doc-commands (backend and frontend changes)
* limited the size of produced json at the maximum
* API accepts "?pretty" parameter

### CODE REVIEW

File: src/web/api_model.nit [hunk 1]
```diff
@@ -248,7 +248,8 @@ class APIEntityDefs
 		mentities = filter_mentities(req, mentities)
 		mentities = sort_mentities(req, mentities)
 		mentities = limit_mentities(req, mentities)
-		res.json new JsonArray.from(mentities)
+		res.api_json(req, new JsonArray.from(mentities))
+	end
```
Comment (privat) [inline src/web/api_model.nit]: i'm not sure about this double `end`

**Comment by Morriar** [comment]
test this pleasse

**Comment by Morriar** [comment]
test this please

**Comment by Morriar** [comment]
test this please

**Comment by privat** [comment]
still some errors

> src/doc/api/api_graph.nit:46,7--22: Redef Error: `InheritanceGraph` is an imported class. Add the `redef` keyword to refine it.

**Comment by Morriar** [comment]
Reroll:
* rebased on master to save conflicts with new nitlight
* removed the extraneous `api_graph` module

**Pull request merged**